src/validation.h

```diff
@@ -430,6 +427,11 @@ class BlockManager
       */
     std::set<CBlockIndex*> m_failed_blocks;
 
+    CBlockIndex* pindexBestInvalid = nullptr;
```

Comment:
I believe they are accessed by `BlockManager` here: https://github.com/bitcoin/bitcoin/blob/21438d55d553ae5bf3be7c0d4431aaf136db6c6b/src/validation.cpp#L3732-L3737

Additionally (and I'd like feedback on if this makes sense), it seems to me that variables whose type involve `CBlockIndex*` should likely live an die with the `BlockManager` since it owns the `BlockMap` entries to which these pointers are pointing.